compiler/AST/primitive.cpp

```diff
@@ -702,11 +702,12 @@ initPrimitive() {
   prim_def(PRIM_REQUIRE, "require", returnInfoVoid, false, false);
 }
 
-Map<const char*, VarSymbol*> memDescsMap;
+static Map<const char*, VarSymbol*> memDescsMap;
+static Map<int, VarSymbol*> memDescsNodeMap;
```

Comment:
I don't have a strong opinion about this. However I think we should be clear with ourselves whether or not Type* can be deleted. E.g. in LLVM they cannot. I sense you prefer hashing on integer IDs which is OK for now.